src/main/ml-modules/root/server/endpoints/adhoc/wizard/api-adhoc-query-wizard-doc-selection.xqy

```diff
@@ -0,0 +1,138 @@
+xquery version "1.0-ml";
+
+import module namespace functx = "http://www.functx.com" at "/MarkLogic/functx/functx-1.0-nodoc-2007-01.xqy";
+import module namespace cfg = "http://www.marklogic.com/data-explore/lib/config"
+  at "/server/lib/config.xqy";
+import module namespace const = "http://www.marklogic.com/data-explore/lib/const"
+at "/server/lib/const.xqy";
+import module namespace lib-adhoc = "http://marklogic.com/data-explore/lib/adhoc-lib" at "/server/lib/adhoc-lib.xqy";
+import module namespace lib-adhoc-create = "http://marklogic.com/data-explore/lib/adhoc-create-lib" at "/server/lib/adhoc-create-lib.xqy";
+import module namespace to-json = "http://marklogic.com/data-explore/lib/to-json" at "/server/lib/to-json-lib.xqy";
+import module namespace xu = "http://marklogic.com/data-explore/lib/xdmp-utils" at "/server/lib/xdmp-utils.xqy"; 
+import module namespace  check-user-lib = "http://www.marklogic.com/data-explore/lib/check-user-lib" at "/server/lib/check-user-lib.xqy" ;
+import module namespace detail-lib = "http://www.marklogic.com/data-explore/lib/detail-lib" at "/server/lib/detail-lib.xqy";
+
+declare function local:get-uris-by-directory($path, $max-uris,$start-uri,$db) {
+let $max-uris:=xs:int($max-uris)
+let $max-uris-plus1:=$max-uris+1
+let $path:=if(fn:ends-with($path,"/")) then $path else $path||"/"
+let $path:=if(fn:starts-with($path,"/")) then $path else "/"||$path
+
+let $query:=<query>
+cts:uris("{$start-uri}",('score-zero',"limit={$max-uris-plus1}"),cts:directory-query("{$path}","infinity"))[2 to {$max-uris-plus1}]
```

Comment:
Any reason why this is serialized first? I think it reads more cleanly like this:

    cts:uris(
      $start-uri, ("score-zero", "limit=" || $max-uris-plus1), 
      cts:directory-query($path, "infinity")
    )[2 to $max-uris-plus1]

Reply:
i have to pass the query string into eval against db passed like this : xu:eval($query....
I figured instead of appending strings the above is cleaner.